src/main/java/com/visenze/productsearch/param/ImageSearchParam.java

```diff
@@ -0,0 +1,257 @@
+package com.visenze.productsearch.param;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Multimap;
+import com.visenze.visearch.Box;
+import com.visenze.visearch.ResponseMessages;
+import com.visenze.visearch.internal.InternalViSearchException;
+
+import java.io.File;
+
+import static com.visenze.visearch.internal.constant.ViSearchHttpConstants.*;
+import static com.visenze.visearch.internal.constant.ViSearchHttpConstants.DETECTION_SENSITIVITY;
+
+/**
+ * <h1> SBI Parameters </h1>
+ * This class holds the specific parameters required to support Search by Image.
+ * It extends the BaseSearchParam to inherit all it's variables that defines all
+ * general search parameter.
+ *
+ * @author Shannon Tan
+ * @version 1.0
+ * @since 08 Jan 2021
+ */
+public class ImageSearchParam extends BaseProductSearchParam {
+
+    /**
+     * Image URL
+     */
+    protected String im_url;
+
+    /**
+     * Image ID
+     */
+    protected String im_id;
+
+    /**
+     * Image file object (multi-part)
+     */
+    protected File image;
+
+    /**
+     * Optional parameter for restricting the image area x1,y1,x2,y2. The
+     * upper-left corner of an image is (0,0).
+     */
+    protected Optional<Box> box = Optional.absent();
+
+    /**
+     * Turn on automatic object detection so the algorithm will try to detect
+     * the object in the image and search.
+     */
+    protected Optional<String> detection = Optional.absent();
+
+    /**
+     * Maximum number of products could be detected for a given image; default
+     * value is 5. Values must range between 1-30. Returns the objects with
+     * higher confidence score first.
+     */
+    protected Optional<Integer> detection_limit = Optional.absent();
+
+    /**
+     * Parameter to set the detection to more or less sensitive. Default is low.
+     */
+    protected Optional<String> detection_sensitivity = Optional.absent();
+
+    /**
+     * Constructor that must have one of the string parameters assigned with
+     * valid values.
+     *
+     *      Both - GOOD but unsure which will be used,
+     *      Only One - GOOD,
+     *      Neither - BAD
+     *
+     * @param image_id image id
+     * @param image_url image url
+     */
+    public ImageSearchParam(String image_url, String image_id) {
```

Comment:
a better way to do this is to have a Builder that take 1 of the 3 param. Alternately, you can have a static method e.g. ImageSearchParam.newFromImageURL(), .newFromImId